crates/bevy_render/src/gpu_readback.rs

```diff
@@ -0,0 +1,341 @@
+use crate::{
+    extract_component::ExtractComponentPlugin,
+    prelude::Image,
+    render_asset::RenderAssets,
+    render_resource::{Buffer, BufferUsages, Extent3d, ImageDataLayout, Texture, TextureFormat},
+    renderer::{render_system, RenderDevice},
+    storage::{GpuShaderStorageBuffer, ShaderStorageBuffer},
+    texture::{GpuImage, TextureFormatPixelInfo},
+    ExtractSchedule, MainWorld, Render, RenderApp, RenderSet,
+};
+use async_channel::{Receiver, Sender};
+use bevy_app::{App, Plugin};
+use bevy_asset::Handle;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::schedule::IntoSystemConfigs;
+use bevy_ecs::{
+    change_detection::ResMut,
+    entity::Entity,
+    event::Event,
+    prelude::{Component, Resource, World},
+    system::{Query, Res},
+};
+use bevy_reflect::Reflect;
+use bevy_render_macros::ExtractComponent;
+use bevy_utils::{default, tracing::warn, HashMap};
+use wgpu::{CommandEncoder, COPY_BYTES_PER_ROW_ALIGNMENT};
+
+const MAX_UNUSED_FRAMES: usize = 10;
```

Comment:
Why `10`?

Reply:
Pretty arbitrary, my local testing showed that there was between 1-3 frames delay typically which was fairly consistent, so you might see 2 buffers unused for 5 frames but then it spikes to 3 again. If anything it might benefit from being longer but the key thing is just to avoid reallocating the same required few buffers over and over again which this def is suitable for. We can tune more in the future or allow user to configure if necessary.

Reply:
I'd go with whatever is more common practice in bevy for now. Thanks for the response!

Reply:
Maybe add a

`// avoid reallocating the same required few buffers over and over again`

Reply:
Added a comment to the const